languages/csharp-dotnet-core2/config.json

```diff
@@ -0,0 +1,6 @@
+{
+    "artifacts": [
+        {"source":"/opt/algorithm/bin/Release/*/*.dll", "destination":"/opt/algorithm/Algorithm.dll"},
```

Comment:
the Wildcards need to remain, otherwise if the user changes the name of the primary namespace, we won't migrate the right files over.

Reply:
Can we hardcode the second `*`  to `Algorithm`  cause we know the class has that name?
This seems strange. are we sure it will target 1 file? if not, then this will throw an error on build cause this has to match one file. Same comment for the next one

Reply:
We don't know if that class will remain the same, the user can change the names. This resolves IFF there is only one compiled artifact. If we set the gitignore to ignore the `bin` and `obj` directories, we can ensure this is always the case (unless the user messes with the gitignore). I can verify that if there's a single file with that pattern, it works properly.

So the question here is: Do we want to force users to never change the primary namespace, or should we force them to not delete the gitignore and push build products to git (which breaks things)?

The former is simpler for us but restricts the user a bit, the latter might be solvable with some fancy shell scripting

Reply:
I added the build directories to a .gitignore in `template` which should avoid duplicate build products - although I agree it's a little brittle still. I'd be ok with having a discussion around this, but I'd also like to ship something to vistaprint asap and I think this would be a minor pain in the grand scheme...

Reply:
Why don't we merge this, keep it experimental and, if we come up with a better solution eventually, We swap to that before a full rollout?

Reply:
Solutions made - we now move the entire compile directory into `/opt/algorithm`, and run `dotnet *.dll`. This means that we get the best of both worlds, in that there's no brittleness for multiple compiles (it'll still work) - and we can let users change the primary namespace / build products without breaking anything.